nomos-services/sdp/src/adapters/wallet/mock.rs

```diff
@@ -0,0 +1,108 @@
+use std::convert::Infallible;
+
+use ed25519_dalek::{Signer as _, SigningKey};
+use nomos_core::{
+    mantle::{NoteId, Op, OpProof, SignedMantleTx, Transaction as _, tx_builder::MantleTxBuilder},
+    proofs::zksig::{DummyZkSignature, ZkSignaturePublic},
+    sdp::{ActiveMessage, DeclarationMessage, WithdrawMessage, ZkPublicKey},
+};
+
+use crate::adapters::wallet::SdpWalletAdapter;
+
+pub struct MockWalletAdapter;
+
+impl SdpWalletAdapter for MockWalletAdapter {
+    type Error = Infallible;
+
+    fn new() -> Self {
+        Self {}
+    }
+    fn declare_tx(
+        &self,
+        tx_builder: MantleTxBuilder,
+        declaration: Box<DeclarationMessage>,
+    ) -> Result<SignedMantleTx, Self::Error> {
+        // todo: this is for mock, we need signing key in production
+        let signing_key = SigningKey::from_bytes(&[0u8; 32]);
```

Comment:
Where do we hold the pk for this signing key? in #1840 @davidrusu prepared so we can sign stuff.